src/main/scala/shine/DPIA/Phrases/Phrase.scala

```diff
@@ -47,9 +47,9 @@ final case class DepLambda[T, I, U <: PhraseType](kind: Kind[T, I],
 }
 
 object DepLambda {
-  def apply[T, I](kind: Kind[T, I], x: I): Object {
-    def apply[U <: PhraseType](body: Phrase[U]): DepLambda[T, I, U]
-  } = new {
+  def apply[T, I](kind: Kind[T, I], x: I): SyntaxHelper[T, I] = SyntaxHelper(kind, x)
+
+  case class SyntaxHelper[T, I](kind: Kind[T, I], x: I) {
     def apply[U <: PhraseType](body: Phrase[U]): DepLambda[T, I, U] = DepLambda(kind, x, body)
   }
 }
```

Comment:
Can we use a `case class DepLambda` to avoid the helper?

Reply:
Of course, there is a class `DepLambda` that represents a dependent lambda phrase.
We usually keep those classes free of any syntactic sugar (which I think is a good idea) and even for this here it is not clear if this syntax should be defined here or elsewhere in a DSL package.

Anyway, I think the purpose of this is only to infer the type `U` and not have to annotate this, while still allowing the annotation of `T` and `I`.

We might want to look into this after having migrated to Scala 3 again, I supose ...